instrumentation/vibur-dbcp-11.0/testing/src/main/java/io/opentelemetry/instrumentation/viburdbcp/AbstractViburInstrumentationTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.viburdbcp;
+
+import static org.mockito.Mockito.when;
+
+import io.opentelemetry.instrumentation.testing.internal.AutoCleanupExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.db.DbConnectionPoolMetricsAssertions;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.assertj.core.api.AbstractIterableAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.vibur.dbcp.ViburDBCPDataSource;
+
+@ExtendWith(MockitoExtension.class)
+public abstract class AbstractViburInstrumentationTest {
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  @Mock DataSource dataSourceMock;
+  @Mock Connection connectionMock;
+
+  protected abstract InstrumentationExtension testing();
+
+  protected abstract void configure(ViburDBCPDataSource viburDataSource);
+
+  @Test
+  void shouldReportMetrics() throws SQLException, InterruptedException {
+    // given
+    when(dataSourceMock.getConnection()).thenReturn(connectionMock);
+
+    ViburDBCPDataSource viburDataSource = new ViburDBCPDataSource();
+    viburDataSource.setExternalDataSource(dataSourceMock);
+    viburDataSource.setName("testPool");
+    configure(viburDataSource);
+    viburDataSource.start();
+
+    // when
+    Connection hikariConnection = viburDataSource.getConnection();
+    TimeUnit.MILLISECONDS.sleep(100);
+    hikariConnection.close();
+
+    // then
+    DbConnectionPoolMetricsAssertions.create(
+            testing(), "io.opentelemetry.viburdbcp-11.0", "testPool")
+        .disableMinIdleConnections()
+        .disableMaxIdleConnections()
+        .disablePendingRequests()
+        .disableConnectionTimeouts()
+        .disableCreateTime()
+        .disableWaitTime()
+        .disableUseTime()
+        .assertConnectionPoolEmitsMetrics();
+
+    // when
+    // this one too shouldn't cause any problems when called more than once
+    viburDataSource.close();
+    viburDataSource.close();
+
+    // sleep exporter interval
+    Thread.sleep(100);
+    testing().clearData();
+    Thread.sleep(100);
+
+    // then
+    testing()
+        .waitAndAssertMetrics(
+            "io.opentelemetry.hikaricp-3.0",
```

Comment:
how is this passing? 🤔

Reply:
Thanks. Because the intent seems to be to assert that no metrics are received after `close` is called.